contracts/Bridge/BaseXVSProxyOFT.sol

```diff
@@ -0,0 +1,356 @@
+     */
+    mapping(uint16 => uint256) public chainIdToMaxSingleReceiveTransactionLimit;
+    /**
+     * @notice Maximum daily limit for receiving transactions in USD(scaled with 18 decimals) from remote chain.
+     */
+    mapping(uint16 => uint256) public chainIdToMaxDailyReceiveLimit;
+    /**
+     * @notice Total received amount in USD(scaled with 18 decimals) within the last 24-hour window from remote chain.
+     */
+    mapping(uint16 => uint256) public chainIdToLast24HourReceived;
+    /**
+     * @notice Timestamp when the last 24-hour window started from remote chain.
+     */
+    mapping(uint16 => uint256) public chainIdToLast24HourReceiveWindowStart;
+    /**
+     * @notice Address on which cap check and bound limit is not appicable.
+     */
+    mapping(address => bool) public whitelist;
+
+    /**
+     * @notice Emmited when address is added to whitelist.
+     */
+    event SetWhitelist(address indexed addr, bool isWhitelist);
+    /**
+     * @notice  Emitted when the maximum limit for a single transaction from local chain is modified.
+     */
+    event SetMaxSingleTransactionLimit(uint16 chainId, uint256 oldMaxLimit, uint256 newMaxLimit);
+    /**
+     * @notice Emitted when the maximum daily limit of transactions from local chain is modified.
+     */
+    event SetMaxDailyLimit(uint16 chainId, uint256 oldMaxLimit, uint256 newMaxLimit);
+    /**
+     * @notice Emitted when the maximum limit for a single receive transaction from remote chain is modified.
+     */
+    event SetMaxSingleReceiveTransactionLimit(uint16 chainId, uint256 oldMaxLimit, uint256 newMaxLimit);
+    /**
+     * @notice Emitted when the maximum daily limit for receiving transactions from remote chain is modified.
+     */
+    event SetMaxDailyReceiveLimit(uint16 chainId, uint256 oldMaxLimit, uint256 newMaxLimit);
+    /**
+     * @notice Event emitted when oracle is modified.
+     */
+    event OracleChanged(address indexed oldOracle, address indexed newOracle);
+    /**
+     * @notice Event emitted when trusted remote sets to empty.
+     */
+    event TrustedRemoteRemoved(uint16 chainId);
+
+    /**
+     * @param tokenAddress_ Address of the inner token.
+     * @param sharedDecimals_ No of shared decimals.
+     * @param lzEndpoint_ Address of the layer zero endpoint contract.
+     * @param oracle_ Address of the price oracle.
+     * @custom:error ZeroAddressNotAllowed is thrown when token contract address is zero.
+     * @custom:error ZeroAddressNotAllowed is thrown when lzEndpoint contract address is zero.
+     * @custom:error ZeroAddressNotAllowed is thrown when oracle contract address is zero.
+     */
+    constructor(
+        address tokenAddress_,
+        uint8 sharedDecimals_,
+        address lzEndpoint_,
+        address oracle_
+    ) BaseOFTV2(sharedDecimals_, lzEndpoint_) {
+        ensureNonzeroAddress(tokenAddress_);
+        ensureNonzeroAddress(lzEndpoint_);
+
+        innerToken = IERC20(tokenAddress_);
```

Comment:
Do we need an event for state set in the constructor?

Reply:
[aba9539](https://github.com/VenusProtocol/token-bridge/pull/1/commits/aba9539f61aa34f2189e28601e2e2bbcd09166b0)